src/proxy/inbound.rs

```diff
@@ -266,16 +271,28 @@ impl Inbound {
                 }
                 // Orig has 15008, swap with the real port
                 let conn = rbac::Connection { dst: addr, ..conn };
-                let Some(upstream) = workloads.fetch_workload(&addr.ip()).await else {
+                let dst_network_addr = NetworkAddress {
+                    network: conn.dst_network.to_string(), // dst must be on our network
+                    address: addr.ip(),
+                };
+                let Some(upstream) = workloads.fetch_workload(&dst_network_addr).await else {
                     info!(%conn, "unknown destination");
                     return Ok(Response::builder()
                         .status(StatusCode::NOT_FOUND)
                         .body(Empty::new())
                         .unwrap());
                 };
-                let (has_waypoint, from_waypoint) =
-                    Self::check_waypoint(&workloads, &upstream, &conn).await;
+                let has_waypoint = upstream.waypoint.is_some();
+                let from_waypoint = Self::check_waypoint(&workloads, &upstream, &conn)
+                    .await
+                    .unwrap();
```

Comment:
nit: if we are always going to unwrap these -- which looks like its because its unimplemented, may be simpler to do it in the check_gateway_address function. You can do `todo!()`

Reply:
might be useful elsewhere, unless we plan to go awhile with things unimplemented; in that case, we should not unwrap. Basically, proper errors or `todo!()` is preferred. Not a blocker though